File: SDK-64bit/doc/_downloads/almotion_setposition.py
import sys
import time
from naoqi import ALProxy
from naoqi import motion

if (len(sys.argv) < 2):
    print "Usage: 'python motion_setposition.py IP [PORT]'"
    sys.exit(1)

IP = sys.argv[1]
PORT = 9559
if (len(sys.argv) > 2):
    PORT = sys.argv[2]
try:
    proxy = ALProxy("ALMotion", IP, PORT)
except Exception,e:
    print "Could not create proxy to ALMotion"
    print "Error was: ",e
    sys.exit(1)
    
proxy.setStiffnesses("Body", 1.0)
proxy.walkInit()

# Example showing how to set LArm Position, using a fraction of max speed  
chainName = "LArm" 
space     = motion.SPACE_TORSO 
useSensor = False 
 
# Get the current position of the chainName in the same space 
current = proxy.getPosition(chainName, space, useSensor) 
 
target  = [ 
current[0] + 0.1, 
current[1] + 0.1, 
current[2] + 0.1, 
current[3] + 0.0, 
current[4] + 0.0, 
current[5] + 0.0] 
 
fractionMaxSpeed = 0.5 
axisMask         = 7 # just control position 
 
proxy.setPosition(chainName, space, target, fractionMaxSpeed, axisMask)

time.sleep(1.0)

# Example showing how to set Torso Position, using a fraction of max speed
chainName        = "Torso" 
space            = 2 
position         = [0.0, 0.0, 0.25, 0.0, 0.0, 0.0] # Absolute Position
fractionMaxSpeed = 0.2 
axisMask         = 63 
proxy.setPosition(chainName, space, position, fractionMaxSpeed, axisMask)

File: SDK-64bit/doc/_downloads/almotion_setposition.cpp
#include <iostream>
#include <alproxies/almotionproxy.h>
#include <qi/os.hpp>

int main(int argc, char **argv)
{
  if (argc < 2) {
    std::cerr << "Usage: motion_setposition pIp"
              << std::endl;
    return 1;
  }
  const std::string pIp = argv[1];

  AL::ALMotionProxy motion(pIp);

  std::vector<float> stiffnesses = motion.getStiffnesses("Body");
  if (stiffnesses[0] < 0.5f) {
    std::cerr << "Warning: this example will have no effect. "
              << "Robot must be stiff and standing." << std::endl;
    return 1;
  }

  // Example showing how to set Torso Position, using a fraction of max speed
  std::string chainName  = "Torso";
  int space              = 2;
  std::vector<float> position(6, 0.0f); // Absolute Position
  position[2] = 0.25f;
  float fractionMaxSpeed = 0.2f;
  int axisMask           = 63;
  motion.setPosition(chainName, space, position, fractionMaxSpeed, axisMask);
  qi::os::sleep(2.0f);

  return 0;
}
